src/Hardmon.cpp

```diff
@@ -0,0 +1,122 @@
+#include <Hardmon.hpp>
+
+namespace VCU {
+
+Hardmon::Hardmon(reqGPIO gpios, IO::CAN& ptCAN) : gpios(gpios), powertrainCAN(ptCAN) {
+}
+
+CO_OBJ_T* Hardmon::getObjectDictionary() {
+    return &objectDictionary[0];
+}
+
+uint8_t Hardmon::getNumElements() {
+    return OBJECT_DICTIONARY_SIZE;
+}
+
+uint8_t Hardmon::getNodeID() {
+    return NODE_ID;
+}
+
+void Hardmon::handlePowertrainCanMessage(IO::CANMessage& message) {
+    switch (message.getId()) {
+    case DEV::PowertrainCAN::HIB_MESSAGE_ID:
+        forwardEnable = powertrainCAN.parseHIBForwardEnable(message);
+        break;
+    case DEV::PowertrainCAN::UC_SELF_TEST_MESSAGE_ID:
+        powertrainCAN.sendHardmonSelfTestResponse();
+        break;
+    default:
+        //we don't care about this message lol
+        break;
+    }
+}
+
+EVT::core::types::FixedQueue<POWERTRAIN_QUEUE_SIZE, IO::CANMessage>* Hardmon::getPowertrainQueue() {
+    return &powertrainCAN.queue;
+}
+
+void Hardmon::process() {
+    //handle all the powertrain CAN messages
+    IO::CANMessage message;
+    while (!powertrainCAN.queue.isEmpty()) {
+        powertrainCAN.queue.pop(&message);
+        handlePowertrainCanMessage(message);
+    }
+
+    //update inputs
+    //forwardEnable ahs been updated over CAN
+    ignitionCheck = gpios.ignitionCheckGPIO.readPin() == IO::GPIO::State::HIGH;
+    ignition3v3 = gpios.ignition3V3GPIO.readPin() == IO::GPIO::State::HIGH;
+    lvssStatus = gpios.lvssStatus3V3GPIO.readPin() == IO::GPIO::State::HIGH;
+    mcStatus = gpios.mcStatusGPIO.readPin() == IO::GPIO::State::HIGH;
+    ucState[0] = gpios.ucStateZeroGPIO.readPin() == IO::GPIO::State::HIGH;
+    ucState[1] = gpios.ucStateOneGPIO.readPin() == IO::GPIO::State::HIGH;
+    ucState[2] = gpios.ucStateTwoGPIO.readPin() == IO::GPIO::State::HIGH;
+    ucState[3] = gpios.ucStateThreeGPIO.readPin() == IO::GPIO::State::HIGH;
+    eStopCheck = gpios.eStopCheckGPIO.readPin() == IO::GPIO::State::HIGH;
+    //discharge will be updated over CAN
+    watchdog = gpios.watchdogGPIO.readPin() == IO::GPIO::State::HIGH;
+    eStop3v3 = gpios.eStop3V3GPIO.readPin() == IO::GPIO::State::HIGH;
+    //lvssEnableUC should be a pin, but electrical fucked it up and forgot to add it
```

Comment:
What if we didn't curse in our publicly available repositories? Wouldn't that be so epic and cool?

Also, I'd like to hear a bit more about what the issue is here